src/components/Button.tsx

```diff
@@ -29,17 +31,39 @@ type Props = SpacingProps<Theme> &
   VariantProps<Theme, 'buttonVariants'> &
   AllProps<Theme> & {
     // Custom Props...
-    label: string
+    testID?: string
+    label?: string
+    icon?: any
+    textProps?: TextProps<Theme>
   }
 
-export const Button = ({ onPress, label, disabled, ...rest }: Props) => {
+export const Button = ({
+  testID,
+  onPress,
+  label,
+  icon: Icon,
+  activeOpacity,
+  textProps,
+  disabled,
+  ...rest
+}: Props) => {
   const props = useRestyle(restyleFunctions, rest)
 
   return (
-    <TouchableOpacity onPress={onPress} disabled={disabled}>
-      <View {...props}>
-        <Text variant="buttonLabel">{label}</Text>
-      </View>
+    <TouchableOpacity
+      testID={testID}
+      onPress={onPress}
+      disabled={disabled}
+      activeOpacity={activeOpacity}
+    >
+      <Box alignItems="center" justifyContent="center" {...props}>
+        {Icon}
```

Comment:
Doesn't this need to be `<Icon />`?

Reply:
Nope, because we pass an instance of Icon to the component, not a type :)